pipeline-scripts/release.groovy

```diff
@@ -21,24 +23,61 @@ def stageValidation(quay_url, name, advisory) {
         error("Payload ${name} already exists! Cannot continue.")
     }
 
-    // AMH - This may be optional?
-    if (advisory != "") {
-        echo "Verifying advisory exists"
-        res = commonlib.shell(
+    if (!advisory) {
+        echo "Getting current advisory for OCP $version from build data..."
+        res = withEnv(["_VERSION=$version"]) {
+            commonlib.shell(
                 returnAll: true,
-                script: "elliott get ${advisory}"
-        )
+                script: 'elliott --group=openshift-"${_VERSION}" get --json - --use-default-advisory image'
+            )
+        }
+        if(res.returnStatus != 0) {
+            error("🚫 Advisory number for OCP $version couldn't be found from ocp_build_data.")
+        }
+    } else {
+        echo "Verifying advisory ${advisory} exists"
+        res = withEnv(["_VERSION=$version", "_ADVISORY=$advisory"]) {
+            commonlib.shell(
+                returnAll: true,
+                script: 'elliott --group=openshift-"${_VERSION}" get --json - -- "${_ADVISORY}"'
+            )
+        }
 
         if(res.returnStatus != 0){
             error("Advisory ${advisory} does not exist! Cannot continue.")
         }
     }
+
+    def advisoryInfo = readJSON text: res.stdout
+    retval.advisoryInfo = advisoryInfo
+
+    echo "Verifying advisory ${advisoryInfo.id} (https://errata.engineering.redhat.com/advisory/${advisoryInfo.id}) status"
+    if (advisoryInfo.status != 'QE') {
+        error("🚫 Advisory ${advisoryInfo.id} is not in QE state.")
+    }
+    echo "✅ Advisory ${advisoryInfo.id} is in QE state."
+
+    // Extract live ID (if exists) from advisory info
+    // FIXME: Not sure if this is a stable way.
```

Comment:
That seems fine to me. We don't really have a clearer signal from the API and it's unlikely to change.

advisory with live ID:
```
    "errata_id": 2681, 
    "fulladvisory": "RHBA-2019:2681-02", 
    "id": 46049, 
    "old_advisory": "RHBA-2019:46049-02", 
```

advisory without:
```
    "errata_id": 46143, 
    "fulladvisory": "RHBA-2019:46143-01", 
    "id": 46143, 
    "old_advisory": null, 
```

Reply:
I plan to talk with Errata developers next week and see if they have better ideas.